app/controllers/staff/submissions_controller.rb

```diff
@@ -10,7 +12,17 @@ def show
     respond_to do |format|
       format.html
       format.csv do
-        response.headers["Content-Disposition"] = "attachment; filename=\"#{ERB::Util.url_encode(@submission.description)}.csv\""
+        fund = @submission.fund
+        @projects = Activity.project.where(organisation: @submission.organisation).select { |activity| activity.associated_fund == fund }
```

Comment:
It's a pity that `Activity` and `Submission` aren't yet associated, as it would avoid the inefficient `select`:

```ruby
@projects = @submission.activities.project
@third_party_projects = @submission.activities.third_party_project
```

or, if you don't care about the ordering:

```ruby
@projects = @submission.activities.where(level: [:project, :third_party_project])
```

Reply:
Can this be improved now that #555 has landed?

Reply:
@jcoglan Sadly not - that's associating transactions to submissions not activities.